src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -1073,12 +1126,12 @@ private MethodSymbol PerformForEachPatternOverloadResolution(TypeSymbol patternT
                     result = null;
                 }
             }
-            else if (overloadResolutionResult.Results.Length > 1)
+            else if (overloadResolutionResult.GetAllApplicableMembers() is var applicableMembers && applicableMembers.Length > 1)
```

Comment:
@333fred asked me to change this:
https://github.com/dotnet/roslyn/pull/43050#discussion_r412367954

Reply:
We probably don't want to change the behavior in a non-extension scenario. Let's identify a scenario/test that differs based on this.

Reply:
This would be an existing scenario.

```csharp
using System.Threading.Tasks;
using System.Collections.Generic;
class C
{
    public static async Task Main()
    {
        await foreach (var i in new C())
        {
        }
    }
    public IEnumerator<int> GetAsyncEnumerator(int a) => throw null;
    public IEnumerator<int> GetAsyncEnumerator(bool a) => throw null;
}
```

Currently it has the following diagnostics:

```
                    // (8,33): warning CS0278: 'C' does not implement the 'collection' pattern. 'C.GetAsyncEnumerator(int)' is ambiguous with 'C.GetAsyncEnumerator(bool)'.
                    //         await foreach (var i in new C())
                    Diagnostic(ErrorCode.WRN_PatternIsAmbiguous, "new C()").WithArguments("C", "collection", "C.GetAsyncEnumerator(int)", "C.GetAsyncEnumerator(bool)").WithLocation(8, 33),
                    // (8,33): error CS8411: Asynchronous foreach statement cannot operate on variables of type 'C' because 'C' does not contain a suitable public instance or extension definition for 'GetAsyncEnumerator'
                    //         await foreach (var i in new C())
                    Diagnostic(ErrorCode.ERR_AwaitForEachMissingMember, "new C()").WithArguments("C", "GetAsyncEnumerator").WithLocation(8, 33)
```

This change removes the warning, which I believe is correct, because it's nothing to do with ambiguity. Both methods fail overload resolution.

It still errors, so this is not a language change.

Reply:
I've added tests for this case.